### ISSUE TITLE
about css two teacher 

### ISSUE THREAD

**Issue by 156aasdfg** [yuleiniu/introd#2]
Hello, Yu Lei, I would like to ask whether there should be two teacher models for the CSS part, and then predict the mixing. It seems that I don't see this part of the code in the code. Maybe I didn't notice. Can you tell me what this part of the code is.

**Comment by 156aasdfg** [comment]
你好玉磊，我想请问下关于css部分，是否应该有两个老师模型，然后进行预测的混合，在代码里似乎没有看到这部分的代码。或许是我没有注意到，可以请问下这部分代码是什么吗。

**Comment by yuleiniu** [comment]
The ID teacher and OOD teacher share the same parameters. ID teacher is the ensemble model that combines UpDn model and bias model, while OOD teacher is the UpDn model. 
The ID teacher's output `logits_all` is obtained via
https://github.com/yuleiniu/introd/blob/a40407c7efee9c34e3d4270d7947f5be2f926413/css/base_model_introd.py#L59
The OOD teacher's output `logits` is obtained via
https://github.com/yuleiniu/introd/blob/a40407c7efee9c34e3d4270d7947f5be2f926413/css/base_model_introd.py#L56
The knowledge blending is processed via
https://github.com/yuleiniu/introd/blob/a40407c7efee9c34e3d4270d7947f5be2f926413/css/train_introd.py#L79
where we mix two losses as the implementation, which works the same as knowledge blending.
Hope these answer your questions.
